src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -553,6 +553,76 @@ static Value builtinStringReplace(ExecutionState& state, Value thisValue, size_t
     }
 }
 
+static Value builtinStringReplaceAll(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    if (thisValue.isUndefinedOrNull()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().object.string(), true, state.context()->staticStrings().replace.string(), ErrorObject::Messages::GlobalObject_ThisUndefinedOrNull);
+    }
+    Value searchValue = argv[0];
+    Value replaceValue = argv[1];
+    // If searchValue is neither undefined nor null, then
+    if (!searchValue.isUndefinedOrNull()) {
+        // If isRegExp is true, then
+        if (searchValue.isObject() && searchValue.asObject()->isRegExp(state)) {
+            Value flags = searchValue.asObject()->get(state, ObjectPropertyName(state, state.context()->staticStrings().flags)).value(state, searchValue);
+            if (!flags.toString(state)->contains("g")) {
+                ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().String.string(), true, state.context()->staticStrings().match.string(), ErrorObject::Messages::GlobalObject_ThisUndefinedOrNull);
```

Comment:
> Let flags be ? Get(searchValue, "flags").
Perform ? RequireObjectCoercible(flags).
If ? ToString(flags) does not contain "g", throw a TypeError exception.

From the standard, it seems that `RequireObjectCoercible(flags)` is missing.
We should check if `flags` value is undefined or null right before `toString`.

Lastly, throwBuiltinError arguments related with error message looks inadequate.
What about fixing it as follow?
```suggestion
            Value flags = searchValue.asObject()->get(state, ObjectPropertyName(state, state.context()->staticStrings().flags)).value(state, searchValue);
            if (!flags.toString(state)->contains("g")) {
                ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().String.string(), true, state.context()->staticStrings().replaceAll.string(), ErrorObject::Messages::GlobalObject_IllegalFirstArgument);
```

Other looks great 👍

Reply:
Thank you !